avro/src/main/java/com/fasterxml/jackson/dataformat/avro/schema/AvroSchemaHelper.java

```diff
@@ -196,6 +196,6 @@ protected static String getTypeId(JavaType type) {
         if (type.isPrimitive()) {
             return ClassUtil.wrapperType(type.getRawClass()).getName();
         }
-        return type.toCanonical();
+        return type.getRawClass().getName();
```

Comment:
`toCanonical` includes generic type information in `<>`, which the Apache implementation cannot understand and will cause exceptions to be thrown.